### PR TITLE
fix #22

### LINKED ISSUE #22: refactoring: tests setup and teardown
names are for readability, if you have to look at the function defintion to undertstand what it does - it's bad naming

_Originally posted by @foofaev in https://github.com/alex-popov-tech/backend-project-lvl4/pull/16#discussion_r586224971_

Please, consider using migrate + rollback instead of trancating every table
Example - https://github.com/hexlet-boilerplates/fastify-nodejs-application/blob/fc27da44dc881721c84365cb55d4cc3e40e8ea30/__tests__/users.test.js#L20-L26

### PR DESCRIPTION
**Pull request by alex-popov-tech**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 setup:
-	rm -rf node_modules; npm install
+	rm -rf node_modules; rm package-lock.json; npm install
```
Comment (foofaev) [inline Makefile:2]: Why we are removing lock?
Where is a guy who recomends not to do that
https://betterprogramming.pub/npm-ci-vs-npm-install-which-should-you-use-in-your-node-js-projects-51e07cb71e26?gi=2545aa193f4

**Review by foofaev — CHANGES_REQUESTED** [review]

File: Makefile (hunk 1 above)
Comment (alex-popov-tech) [reply Makefile]: okay, done

**Review by foofaev — CHANGES_REQUESTED** [review]
валится линтер

**Comment by alex-popov-tech** [comment]
done

**Comment by foofaev** [comment]
waiting for https://github.com/alex-popov-tech/backend-project-lvl4/pull/26

**Review by foofaev — CHANGES_REQUESTED** [review]
It would be better to update branch with upstream.
Also it's a little confusing to have rollback in beforeEach. On the first iteration it's meaningless.

**Comment by alex-popov-tech** [comment]
maybe it does nothing, but it looks easier and cleaner in spec - just clean whatever there is in before each test

**Comment by foofaev** [comment]
Before functions are called "test setup". It's not common to have test teardown inside test setup. It will confuse your teammates.

PR is empty right now

**Comment by alex-popov-tech** [comment]
i believe its a personal preference, not a standardization...for me its perfectly fine to have tests 'start from empty state' because:
- that makes tests resistant to any broken/not properly app state
- that is common to have pre-conditions in manual tests which clear app state, but there almost no cases with post conditions
- having one 'start from scratch' hook is simpler than having multiple

clear just wipes out app the easiest way possible

**Review by foofaev — APPROVED** [review]

**Pull request merged**